### ISSUE TITLE
Oversubscription of resources in backfill mode

### ISSUE THREAD

**Issue by nikosT** [oar-team/oar3#61]
There is an issue when the backfill occurs and this bug is probably happening when more than one job can be put on scheduling holes. I can confirm that the following/similar experiment in the Nancy-grvingt and Grenoble-dahu clusters leads to oversubscription, as seen in the Gantt diagram. I also attach the jobs as they were extracted from the OAR3 database  [oar_10.csv](https://github.com/user-attachments/files/16659902/oar_10.csv).

![oar_10b (1)](https://github.com/user-attachments/assets/518a223b-a915-4efa-a354-8de7799790ba)

You can reproduce the example by `oarsub` the following jobs:
```
mg.E 64 exclusive 00:05:33
cg.D 32 exclusive 00:09:34
ft.D 64 exclusive 00:03:40
ft.D 64 exclusive 00:03:40
cg.D 32 exclusive 00:09:34
bt.D 121 exclusive 00:09:24
ep.D 16 exclusive 00:02:34
ft.D 64 exclusive 00:03:40
cg.D 32 exclusive 00:09:34
lu.D 64 exclusive 00:11:32
bt.D 121 exclusive 00:09:24
sp.D 121 exclusive 00:09:23
ft.D 64 exclusive 00:03:40
sp.D 121 exclusive 00:09:23
ft.D 64 exclusive 00:03:40
ep.D 16 exclusive 00:02:34
mg.E 64 exclusive 00:05:33
bt.D 121 exclusive 00:09:24
ep.D 16 exclusive 00:02:34
```
A similar dataset was given as input on a 10-node cluster (8 compute nodes, 1 server node, and 1 frontend node). Headers are: app, procs, type, walltime.
The exclusive here means a full node capacity. For instance, the first NPB which needs 64 jobs is equal to `-l /nodes=2` etc.

I'll try to reproduce it on the docker environment. Meanwhile, could you also confirm that on Grid5000?

**Comment by nikosT** [comment]
Hi @augu5te, I managed to reproduce the issue in the docker environment. Assume a 10 node cluster (8 compute nodes, 1 server node, and 1 frontend node). Try to submit this dataset:
```
oarsub -l /nodes=4,walltime=00:10:00 'sleep 599';
oarsub -l /nodes=8,walltime=00:10:00 'sleep 59';
oarsub -l /nodes=4,walltime=00:01:00 'sleep 59';
oarsub -l /nodes=4,walltime=00:01:00 'sleep 59';
oarsub -l /nodes=4,walltime=00:01:00 'sleep 59';
oarsub -l /nodes=4,walltime=00:01:00 'sleep 59';
oarsub -l /nodes=4,walltime=00:01:00 'sleep 59';
```
you will notice that the last 3 jobs are overlapping over the same nodes. I also attach some screenshots from Drawgantt and Monika:
![Screenshot at 2024-08-20 17-38-30](https://github.com/user-attachments/assets/299bc482-5879-495e-b967-45063a4d3841)
![Screenshot at 2024-08-20 17-38-49](https://github.com/user-attachments/assets/c9c8fd91-6e73-4195-b515-0f8ac4ff7976)

**Comment by nikosT** [comment]
The problem is in the slotset indeed, as @augu5te mentioned. I have created a branch with one commit (currently) ahead of oar-team/oar3:master that seems to solve the issue. I rolled back to a previous commit and made some necessary changes to adjust to new code. 
Actually, I do not consider it as a clean commit, that's why I didn't PR it directly to oar3. We then need to clean it up and keep only the necessary changes. Perhaps, not all files are needed. 
Affected files are one commit before: https://github.com/nikosT/oar3/commit/eb2af22dea7dfa060ce5cc2f03a51966c0bffe49

See branch at: https://github.com/nikosT/oar3/tree/slotset_quick_fix
